src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -226,7 +226,7 @@ pub const KATA_ANNO_CFG_HYPERVISOR_MEMORY_PREALLOC: &str =
     "io.katacontainers.config.hypervisor.enable_mem_prealloc";
 /// A sandbox annotation to specify if the memory should be pre-allocated from huge pages.
 pub const KATA_ANNO_CFG_HYPERVISOR_HUGE_PAGES: &str =
-    "io.katacontainers.config.hypervisor.enable_hugepages";
+    "io.katacontainers.config.hypervisor.hugepages";
```

Comment:
It's called `enable_hugepages` [in the Go version](https://github.com/kata-containers/kata-containers/blob/3d8688f92e68630b15d1bf8b4888857fd8d472b1/src/runtime/virtcontainers/pkg/annotations/annotations.go#L170). What is the rationale for having a different name in the Rust version?

Reply:
I agree with this comment, I think we should keep backward compatible in Rust runtime.
@c3d has presented an easy way to do so.

Reply:
> It's called `enable_hugepages` [in the Go version](https://github.com/kata-containers/kata-containers/blob/3d8688f92e68630b15d1bf8b4888857fd8d472b1/src/runtime/virtcontainers/pkg/annotations/annotations.go#L170). What is the rationale for having a different name in the Rust version? I understand that you have a string now, but I'm concerned that someone passing `enable_hugepages=true` would get no effect with the Rust runtime.
> 
> My recommendation would be to accept both `enable_hugepages` and `hugepages`, and at least for the latter, accept `true` (same as `hugetlbfs`) and `false` as well as `thp`, `hugetlbfs` and `none` (same as `false`). This would provide a behaviour that is compatible both with your current preference as well as with historical input.
>
Using both `enable_hugepages` and other parameters such as `hugepages` at the same time may also lead to user misconfiguration. Following the usage habit of the kata in golang, only configuring `enable_hugepages` but forgetting to configure `hugepages` may also result in the functionality not achieving the expected effect.

So, I agree with the idea that `hugepages` with Enum and make it compatible with both current and historic input.

Reply:
@Apokleos  I think it is hard to provide a behaviour that keep backward compatible in Rust runtime in one parameter. In the current version , enable_hugepages is bool type(true and false) but the new parameter should provide string type("thp" "hugetlbfs"). These two types cannot be configured as one parameter.
IMO we need two parameter to make it compatible with both current and historic input.

Reply:
> @Apokleos I think it is hard to provide a behaviour that keep backward compatible in Rust runtime in one parameter. In the current version , enable_hugepages is bool type(true and false) but the new parameter should provide string type("thp" "hugetlbfs"). These two types cannot be configured as one parameter. IMO we need two parameter to make it compatible with both current and historic input.

Since we parse the text input, we can parse `true` as meaning `hugetlbfs`.

Reply:
> Using both `enable_hugepages` and other parameters such as `hugepages` at the same time may also lead to user misconfiguration. Following the usage habit of the kata in golang, only configuring `enable_hugepages` but forgetting to configure `hugepages` may also result in the functionality not achieving the expected effect.

What kind of misconfiguration are you thinking about? To me, this is really an optimization in how you allocate memory. If you say "enable hugepages" and get the same thing as before, but not the new fancy stuff, that's not misconfiguration. It's not a problem that you need to say something new to enable the new option. So I do not understand the objection.